src/sentry/interfaces/stacktrace.py

```diff
@@ -353,16 +353,20 @@ def to_python(cls, data, raw=False):
 
         # XXX: handle lines which were sent as 'null'
         context_line = trim(data.get('context_line'), 256)
-        if context_line is not None:
-            pre_context = data.get('pre_context', None)
-            if pre_context:
-                pre_context = [c or '' for c in pre_context]
-
-            post_context = data.get('post_context', None)
-            if post_context:
-                post_context = [c or '' for c in post_context]
+        pre_context = data.get('pre_context', None)
+        if isinstance(pre_context, list) and pre_context:
+            pre_context = [c or '' for c in pre_context]
         else:
-            pre_context, post_context = None, None
+            pre_context = None
+
+        post_context = data.get('post_context', None)
+        if isinstance(post_context, list) and post_context:
+            post_context = [c or '' for c in post_context]
+        else:
+            post_context = None
+
+        if not context_line and (pre_context or post_context):
```

Comment:
Should this not be `pre_context is not None or post_context is not None`?

Reply:
any falsy pre_context/post_context is coerced to `None` before

Reply:
Ah yep.